### DISCUSSION TITLE
Testing for styles. Error: queryByTestId is not defined

### DISCUSSION THREAD

**Discussion by MeenaFoerderer** [neuefische/questions#413] [Web]
Hello,
I want to complete my tests and also test the style (background color) of an element of my component. I followed a suggestion on stackoverflow to add a `data-testid` to the element and test with `queryByTestId`. However, this doesn't work. I get the error that queryByTestId is not defined. All the other tests I did so far for this component work fine. Why does this fail?

The element (a styled div) in my component:
![talkcard js](https://user-images.githubusercontent.com/105375266/208250103-b516410e-aab9-4838-b242-542822d1d85b.jpg)

My test.js file:

![talkcard test1](https://user-images.githubusercontent.com/105375266/208250187-044d4ac8-fe34-4dbd-a89e-6718ea7f67dc.jpg)
![talkcard test2](https://user-images.githubusercontent.com/105375266/208250194-76d50edc-804a-463e-9775-d0fde1b027ee.jpg)

Error in my console:
![console](https://user-images.githubusercontent.com/105375266/208250213-7eb9b858-e29a-4d9a-904d-1a45843ac2c6.jpg)

**Comment by stefankorder** [comment]
Hi Meena,
probably because `queryByTestId` doesn't exist like that. Usually you access the properties of `screen` or import the function. See also: https://testing-library.com/docs/queries/bytestid/
